Adding custom move feedrate for G26

ScrewThisBanana opened the pull request:
### Requirements

`G26_MESH_VALIDATION` enabled in configuration, no additional requirements
<!--* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.-->

### Description

This commit adds an additional configuration parameter (`G26_XY_FEEDRATE_MOVE`) in the Configuration.h file that can be used to specify the movement speed during the G26 validation pattern command during moves without extrusion.

The added feature was tested on my personal printer (TwoTrees Sapphire Pro) and works like a charm. Only x/y movements without e components take the newly introduced feedrate. All other behaviours of the G26 command stay the same.
<!--

We must be able to understand your proposed change from this description. If we can't understand what the code will do from this description, the Pull Request may be closed at the maintainers' discretion. Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code recently, so please walk us through the concepts.

-->

### Benefits

<!-- What does this fix or improve? -->

The amount of time used for G26 is drastically reduced as the feedrate for print moves is now different from moves without E-components

### Configurations

<!-- Attach any Configuration.h, Configuration_adv.h, or platformio.ini files needed to compile/test your Pull Request. -->

[platformio.ini.txt](https://github.com/MarlinFirmware/Marlin/files/5791247/platformio.ini.txt)
[Configuration.h.txt](https://github.com/MarlinFirmware/Marlin/files/5791251/Configuration.h.txt)

### Related Issues

<!-- Whether this fixes a bug or fulfills a feature request, please list any related Issues here. -->

Closes MarlinFirmware/Marlin#20615

Commit:
Adding custom move feedrate for G26

This commit adds an additional configuration parameter that can be used to specify the movement speed during the G26 validation pattern command during moves without extrusion.

Closes MarlinFirmware/Marlin#20615

Commit:
Fixing missing default 'G26_XY_FEEDRATE_MOVE' value

This commit adds a default 'G26_XY_FEEDRATE_MOVE' value (max movement speed / 1.5) in the G26.cpp - same behaviour as the default 'G26_XY_FEEDRATE' value

Commit:
Adding comment describing functionality in G26.cpp

qwewer0 commented:
I think `G26_XY_FEEDRATE_TRAVEL` would reflect the settings more, and maybe `100` would be safer, while still acceptable as default value.
@ScrewThisBanana What do you think?

qwewer0 commented:
Tried it out, works as expected.

ScrewThisBanana commented:
> I think `G26_XY_FEEDRATE_TRAVEL` would reflect the setting more, and maybe `100` would be safer, while still acceptable as default value.
> @ScrewThisBanana What do you think?

I will change that two topics tomorrow

Commit:
Renaming 'G26_XY_FEEDRATE_MOVE' to 'G26_XY_FEEDRATE_TRAVEL'

Configuration parameter renamed for better readability and consistency

MarlinFirmware/Marlin#20615

ScrewThisBanana commented:
> > I think `G26_XY_FEEDRATE_TRAVEL` would reflect the setting more, and maybe `100` would be safer, while still acceptable as default value.
> > @ScrewThisBanana What do you think?
> 
> I will change that two topics tomorrow

"Tomorrow" is relative - fixed now ^^

qwewer0 commented on Marlin/Configuration.h:
```diff
@@ -1361,6 +1361,7 @@
     #define MESH_TEST_HOTEND_TEMP  205    // (°C) Default nozzle temperature for the G26 Mesh Validation Tool.
     #define MESH_TEST_BED_TEMP      60    // (°C) Default bed temperature for the G26 Mesh Validation Tool.
     #define G26_XY_FEEDRATE         20    // (mm/s) Feedrate for XY Moves for the G26 Mesh Validation Tool.
+    #define G26_XY_FEEDRATE_TRAVEL    150   // (mm/s) Feedrate for XY Moves without extrusion for the G26 Mesh Validation Tool
```
@ScrewThisBanana Could you align it, like the values above?

ScrewThisBanana replied:
Ofc

ScrewThisBanana replied:
@qwewer0 done: https://github.com/MarlinFirmware/Marlin/pull/20729/commits/aaa873e4eb6a13d09f16a118d0da551a3f558a00

Commit:
Setting 'G26_XY_FEEDRATE_TRAVEL' to a safer value, aligned comments

Changed default value for 'G26_XY_FEEDRATE_TRAVEL' from 150 mm/s to 100 mm/s for safety purposes, comment alignment

MarlinFirmware/Marlin#20615

qwewer0 commented on Marlin/Configuration.h:
```diff
@@ -1361,6 +1361,7 @@
     #define MESH_TEST_HOTEND_TEMP  205    // (°C) Default nozzle temperature for the G26 Mesh Validation Tool.
     #define MESH_TEST_BED_TEMP      60    // (°C) Default bed temperature for the G26 Mesh Validation Tool.
     #define G26_XY_FEEDRATE         20    // (mm/s) Feedrate for XY Moves for the G26 Mesh Validation Tool.
+    #define G26_XY_FEEDRATE_TRAVEL  100   // (mm/s) Feedrate for XY Moves without extrusion for the G26 Mesh Validation Tool
```
Decimal point should be aligned for all.

ScrewThisBanana replied:
By the way, your change regarding the decimal point alignment is done in the new (needed) PR https://github.com/MarlinFirmware/Marlin/pull/20879 in commit https://github.com/MarlinFirmware/Marlin/pull/20879/commits/5d930a35753b399ba5234cbb19a77f29dde10858 - I finally understood what you wanted from me ;) I hope that is ok for you?

qwewer0 replied:
All ok, thank you.

Roxy-3D commented:
@ScrewThisBanana    This is a long overdue improvement.   

Thank You!

Roxy-3D commented:
@ScrewThisBanana    Ooooops!   I shouldn't have merged this yet.   We need to keep all the example Configuration.h files synchronized with the default Configuration.h file.     I need to revert this.

As soon as the Example Configuration.h files have the new option in them we can continue.

ScrewThisBanana commented:
@Roxy-3D how should I proceed? Pull example configuration files from https://github.com/MarlinFirmware/Configurations and add the new configuration parameter in each example configuration file?

Roxy-3D commented:
@ScrewThisBanana     I apologize!   I seem to be a little bit out of the loop.   What I said used to be very true.   All of the (many) Example Configuration.h files needed to be updated to reflect any new options added.    But that is no longer true.

What you had was sufficient and I should not have reverted the merge.     I don't think there is an easy way for me to recover the original Pull Request.   If you can start a fresh Pull Request with the exact changes reflected up above, I will get it merged again.

ScrewThisBanana commented:
@Roxy-3D A new pull request (https://github.com/MarlinFirmware/Marlin/pull/20729#issuecomment-766449322) was opened, feel free to try again ;)

qwewer0 commented:
@Roxy-3D Can you remove/delete [revert-20729-bugfix-2.0.x branch](https://github.com/MarlinFirmware/Marlin/tree/revert-20729-bugfix-2.0.x)? I think it served it's purpose and will only cause confusion.
